hir/src/locals.rs

```diff
@@ -0,0 +1,86 @@
+use std::{alloc::Layout, fmt};
+
+use super::Type;
+
+/// A strongly typed identifier for referencing locals associated with a function
```

Comment:
Locals are implicitly created by use of the `alloca` instruction, which returns the address of a `Local` as a `Value`, which can then be used with the `load` and `store` instructions to read/write it. There is no way to directly create a `Local`, aside from using `alloca`. We don't actually reify a `Local` until code generation though, since it isn't necessary to represent in HIR, however they have a more important role in MASM.